centos7.sh

```diff
@@ -31,45 +31,60 @@ msg_opt_complete(){
 }
 
 do_install(){
+
+  # practical options for curl. use with care.
+  CURL_OPTS="--insecure"
   # install basic packages
   yum install -y yum-utils device-mapper-persistent-data lvm2
 
-  # install docker-ce and EPEL repos
+  # install docker-ce
   yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
-  yum install -y epel-release
+  # install EPEL using rpm so it works for both RHEL and CentOS
+  yum install -y http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
 
   # update base OS
   yum update -y
 
   # install required packages
+  yum install -y http://mirror.centos.org/centos/7/extras/x86_64/Packages/container-selinux-2.107-3.el7.noarch.rpm
```

Comment:
I'm not a fan of pulling explicit versions via direct URLs. IMHO the better way would be to subscribe the extras repo with `subscription-manager repos --enable=rhel-7-server-extras-rpms` and then `yum install container-selinux` and let yum figure out version/mirror to use. We probably need to detect RHEL vs CentOS earlier in the script and then check various repo's are subscribed (indicating the system is licensed) or bail out.

Reply:
Unfortunately, the version available in the extra resopitory is too old for docker-ce